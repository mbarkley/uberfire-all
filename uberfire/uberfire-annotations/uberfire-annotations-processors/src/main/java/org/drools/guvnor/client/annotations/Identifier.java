package org.drools.guvnor.client.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import javax.inject.Qualifier;

/**
 * An identifier for a Place (WorkbenchPart or Popup)
 */
@Qualifier
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE})
public @interface Identifier {

    String value();

}