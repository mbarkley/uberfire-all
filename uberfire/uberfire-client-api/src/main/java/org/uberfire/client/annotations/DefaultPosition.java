/*
 * Copyright 2012 JBoss Inc
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package org.uberfire.client.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.uberfire.workbench.model.Position;

import static org.uberfire.workbench.model.Position.ROOT;

/**
 * Default position for WorkbenchParts added to the Workbench. Methods marked
 * with this annotation should take zero arguments and return
 * {@code org.drools.guvnor.client.workbench.Position} indicating the
 * WorkbenchPart's default position.
 */
@Inherited
@Retention(RetentionPolicy.RUNTIME)
@Target({ ElementType.METHOD })
public @interface DefaultPosition {

    Position getPosition() default ROOT;

}
