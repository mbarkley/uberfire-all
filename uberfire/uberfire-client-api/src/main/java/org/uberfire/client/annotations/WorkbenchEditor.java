/*
 * Copyright 2012 JBoss Inc
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package org.uberfire.client.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.uberfire.client.workbench.type.ClientResourceType;

/**
 * Classes annotated with this are considered WorkbenchParts that perform some
 * "editor" function for the specified file-type.
 * <p>
 * At its simplest form the Class should implement
 * {@code com.google.gwt.user.client.ui.IsWidget} (e.g. extend
 * {@code com.google.gwt.user.client.ui.Composite}) and provide a method
 * annotated with {@code @WorkbenchPartTitle}.
 * </p>
 * <p>
 * Developers wishing to separate view from logic (perhaps by implementing the
 * MVP pattern) can further provide a zero-argument method annotated with
 * {@code @WorkbenchPartView} with return type
 * {@code com.google.gwt.user.client.ui.IsWidget}.
 * </p>
 * <p>
 * In this latter case the {@code @WorkbenchEditor} need not implement
 * {@code com.google.gwt.user.client.ui.IsWidget}.
 * </p>
 * <p>
 * WorkbechEditors can receive the following life-cycle calls:
 * <ul>
 * <li>{@code @IsDirty}</li>
 * <li>{@code @OnClose}</li>
 * <li>{@code @OnFocus}</li>
 * <li>{@code @OnLostFocus}</li>
 * <li>{@code @OnMayClose}</li>
 * <li>{@code @OnSave}</li>
 * <li>{@code @OnStartup(org.drools.guvnor.vfs.Path)}</li>
 * <li>{@code @OnOpen}</li>
 * </p>
 */
@Inherited
@Retention(RetentionPolicy.RUNTIME)
@Target({ ElementType.TYPE })
public @interface WorkbenchEditor {

    /**
     * Identifier that should be unique within application.
     */
    String identifier();

    /**
     * Array that defines all supported types of this editor.
     */
    Class<? extends ClientResourceType>[] supportedTypes() default { };

    /**
     * Defines the priority of editor over type resolution, editors with same supported type will be resolved by priority.
     */
    int priority() default 0;

    /**
     * Defines the preferred height. Preferred means that this Height will be used only if this screen
     * is the trigger to create a new panel, if panel already exists this information is ignored.
     */
    int preferredHeight() default -1;

    /**
     * Defines the preferred width. Preferred means that this Width will be used only if this screen
     * is the trigger to create a new panel, if panel already exists this information is ignored.
     */
    int preferredWidth() default -1;
}
