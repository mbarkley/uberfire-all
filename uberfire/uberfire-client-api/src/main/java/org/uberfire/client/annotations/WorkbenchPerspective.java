/*
 * Copyright 2012 JBoss Inc
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package org.uberfire.client.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.uberfire.workbench.model.PerspectiveDefinition;

/**
 * Classes annotated with this are considered Perspective providers.
 * <p>
 * Developers will need to provide a zero-argument method annotated with
 * {@code @Perspective} with return type
 * {@code org.uberfire.client.workbench.model.PerspectiveDefinition}.
 * </p>
 * <p>
 * WorkbenchPerspectives can receive the following life-cycle calls:
 * <ul>
 * <li>{@code @OnClose}</li>
 * <li>{@code @OnStartup}</li>
 * <li>{@code @OnOpen}</li>
 * </p>
 */
@Inherited
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE})
public @interface WorkbenchPerspective {

    String identifier();

    boolean isDefault() default false;

    /**
	 * If true (the default), every time this perspective is displayed, it
	 * should start fresh with the {@link PerspectiveDefinition} returned by the
	 * method annotated with {@code @Perspective}. If false, the framework will
	 * store the structure of the perspective (panel arrangements and part
	 * placement as modified by the user opening and closing tabs, dragging
	 * parts around, and resizing split panels) on the server individually for
	 * each user, and use that stored definition in preference to the one
	 * returned by the {@code @Perspective} method.
	 */
    boolean isTransient() default true;
}
