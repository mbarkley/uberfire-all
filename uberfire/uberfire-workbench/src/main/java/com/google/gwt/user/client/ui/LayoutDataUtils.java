/*
 * Copyright 2012 JBoss Inc
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package com.google.gwt.user.client.ui;

import com.google.gwt.user.client.ui.DockLayoutPanel.Direction;

/**
 * Helper class for DockLayoutPanel.LayoutData which is package protected
 */
public class LayoutDataUtils {

    public static class LayoutData {
        public Direction direction;
        public Integer   size;

        LayoutData(final Direction direction,
                   final int size) {
            this.direction = direction;
            this.size = size;
        }
    }

    public static LayoutData getLayoutData(final Widget w) {
        if ( !(w instanceof SimpleLayoutPanel) ) {
            return null;
        }
        final SimpleLayoutPanel slp = (SimpleLayoutPanel) w;
        final Object o = slp.getLayoutData();
        if ( !(o instanceof DockLayoutPanel.LayoutData) ) {
            return null;
        }
        final DockLayoutPanel.LayoutData layoutData = (DockLayoutPanel.LayoutData) o;
        return new LayoutData( layoutData.direction,
                               (int) layoutData.size );
    }

}
