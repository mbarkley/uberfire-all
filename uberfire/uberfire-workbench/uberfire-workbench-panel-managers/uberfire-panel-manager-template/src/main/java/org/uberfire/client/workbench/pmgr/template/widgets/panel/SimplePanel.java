package org.uberfire.client.workbench.pmgr.template.widgets.panel;

import org.uberfire.client.workbench.part.WorkbenchPartPresenter;
import org.uberfire.workbench.model.PartDefinition;

import com.google.gwt.user.client.ui.ResizeComposite;
import com.google.gwt.user.client.ui.SimpleLayoutPanel;

/**
 * An UberFire panel that can contain zero or one parts.
 */
public class SimplePanel
extends ResizeComposite {

    private final SimpleLayoutPanel container = new SimpleLayoutPanel();

    private PartDefinition partDefinition;

    public SimplePanel() {
        initWidget( container );
    }

    /**
     * Makes the given view fill this panel. Replaces the existing view occupying this panel, if any.
     * 
     * @param part the view this panel should contain. Must not be null.
     */
    public void setPart( final WorkbenchPartPresenter.View part ) {
        clear();
        this.partDefinition = part.getPresenter().getDefinition();
        container.add( part );
    }

    /**
     * Removes the view from this panel, making it empty.
     */
    public void clear() {
        partDefinition = null;
        container.clear();
    }

    /**
     * Returns the PartDefinition of the part that currently occupies this panel.
     * 
     * @return the PartDefinition of the part in this panel, or null if the panel is empty.
     */
    public PartDefinition getPartDefinition() {
        return partDefinition;
    }
}
