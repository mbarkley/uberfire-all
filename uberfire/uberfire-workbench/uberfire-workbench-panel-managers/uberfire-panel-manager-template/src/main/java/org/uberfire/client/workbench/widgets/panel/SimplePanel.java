package org.uberfire.client.workbench.widgets.panel;

import com.google.gwt.user.client.ui.Composite;
import com.google.gwt.user.client.ui.FlowPanel;
import org.uberfire.client.workbench.part.WorkbenchPartPresenter;
import org.uberfire.workbench.model.PartDefinition;

public class SimplePanel
        extends Composite {

    private FlowPanel container = new FlowPanel();

    private PartDefinition partDefinition;

    public SimplePanel() {
        initWidget( container );
    }


    public void setPart( final WorkbenchPartPresenter.View part ) {
        this.partDefinition = part.getPresenter().getDefinition();
        container.add( part );
    }

    public void clear() {
        partDefinition = null;
        container.clear();
    }

}
