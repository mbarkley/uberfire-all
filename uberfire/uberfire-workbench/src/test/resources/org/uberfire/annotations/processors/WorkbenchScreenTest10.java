package org.uberfire.annotations.processors;

import org.uberfire.client.annotations.WorkbenchMenu;
import org.uberfire.client.annotations.WorkbenchPartTitle;
import org.uberfire.client.annotations.WorkbenchPartView;
import org.uberfire.client.annotations.WorkbenchScreen;

import com.google.gwt.user.client.ui.IsWidget;
import com.google.gwt.user.client.ui.SimplePanel;

@WorkbenchScreen(identifier = "test10")
public class WorkbenchScreenTest10 {

    @WorkbenchPartView
    public IsWidget getView() {
        return new SimplePanel();
    }

    @WorkbenchPartTitle
    public String getTitle() {
        return "title";
    }

    @WorkbenchMenu
    public String getMenuBar() {
        return "";
    }

}
