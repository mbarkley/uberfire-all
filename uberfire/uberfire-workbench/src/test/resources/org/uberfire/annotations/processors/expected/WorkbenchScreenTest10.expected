/*
 * Copyright 2012 JBoss Inc
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.uberfire.annotations.processors;

import javax.annotation.Generated;
import javax.enterprise.context.Dependent;
import javax.inject.Inject;
import javax.annotation.PostConstruct;
import org.uberfire.client.mvp.UberView;

import org.uberfire.client.workbench.annotations.Identifier;
import org.uberfire.client.mvp.AbstractScreenActivity;

import com.google.gwt.user.client.ui.IsWidget;

@Dependent
@Generated("org.uberfire.annotations.processors.WorkbenchScreenProcessor")
@Identifier("test10")
/*
 * WARNING! This class is generated. Do not modify.
 */
public class WorkbenchScreenTest10Activity extends AbstractScreenActivity {

    @Inject
    private WorkbenchScreenTest10 realPresenter;

    @PostConstruct
    public void init() {
        ((UberView) realPresenter.getView()).init( realPresenter );
    }

    @Override
    public String getTitle() {
        return realPresenter.getTitle();
    }

    @Override
    public IsWidget getWidget() {
        return realPresenter.getView();
    }
    
}
